tests/models/test_responses.py

```diff
@@ -48,25 +48,15 @@ def test_response_content_type_encoding():
     assert response.encoding == "latin-1"
 
 
-def test_response_autodetect_encoding():
+def test_response_fallback_to_utf8():
     """
-    Autodetect encoding if there is no charset info in a Content-Type header.
-    """
-    content = "おはようございます。".encode("EUC-JP")
-    response = httpx.Response(200, content=content, request=REQUEST)
-    assert response.text == "おはようございます。"
-    assert response.encoding == "EUC-JP"
-
-
-def test_response_fallback_to_autodetect():
-    """
-    Fallback to autodetection if we get an invalid charset in the Content-Type header.
+    Fallback to utf-8 if we get an invalid charset in the Content-Type header.
     """
     headers = {"Content-Type": "text-plain; charset=invalid-codec-name"}
-    content = "おはようございます。".encode("EUC-JP")
+    content = "おはようございます。".encode("utf-8")
     response = httpx.Response(200, content=content, headers=headers, request=REQUEST)
     assert response.text == "おはようございます。"
-    assert response.encoding == "EUC-JP"
+    assert response.encoding == "utf-8"
```

Comment:
So, what if we _do_ get a `"おはようございます。".encode("EUC-JP")`, and the server fails to include `; encoding=EUC-JP`?

I assume we'll try to `.decode("utf-8")` — and that it will fail?

```python
>>> text = "おはようございます。".encode("EUC-JP")
>>> text.decode("utf-8")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xa4 in position 0: invalid start byte
```

In that case I assume we want to allow the user to provide some kind of manual encoding (eg if they _know_ it's some exotic encoding).

Most likely it would just be a recommendation to use something like this...

```python
text = response.content.decode("EUC-JP")
```

This is mostly fine I suppose.

But then, we probably also want to provide more helpful error messaging than the `UnicodeDecodeError` above, correct? For example, a `DecodingError` with an error message saying "consider decoding `.content` manually" (we may have this already, but not sure?).

Reply:
Rather than...

```python
text = response.content.decode("EUC-JP")
print(text)
```

You'd use..

```python
response.encoding = "EUC-JP"
print(response.text)
```